source/manual/aws-iam-key-rotation.html.md

```diff
@@ -5,93 +5,131 @@ layout: manual_layout
+If this user has no keys, consider deleting it if it's not needed.
 
-> **Note**
-> The command you should run is related to the environment you need to target.
-> For example, to edit global staging credentials in the `govuk::apps` namespace, you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.
+If this user has only one key, make a note of the `Access Key ID` and proceed to
+[find out if the key is used].
+
+If this user has two keys, you'll need to follow this guide twice. The first
+time, you'll need to consider one key as the old one to rotate and the other key
+as if it was a new one just created. The second time, you'll follow the guide as
+normal because the user will then have only one key. Now, choose which key is
+the one you are going to rotate and apply the following steps to that key.
+
+### 2. Find out if the key is used
+
+- Check the `Last used` value.
+- If the value is a date, [find where the key is used].
+- If it's `N/A`, you can safely [remove the key] as it was never used.
 
-Now enter [govuk-secrets/puppet_aws](https://github.com/alphagov/govuk-secrets/tree/master/puppet_aws), run the same command again and search for the Access Key ID in the result.
+### 3. Find where the key is used
 
-Keys may also be defined in more then one place, for example the hieradata values
-`govuk::apps::support::aws_access_key_id` and
+- Enter [govuk-secrets/puppet] in your machine.
+- Run the appropriate [Puppet AWS common actions] considering the following:
+  - The command to run is related to the environment you need to target. For
+  example, to edit global staging credentials in the `govuk::apps` namespace,
+  you need to run `bundle exec rake 'eyaml:edit[staging,apps]'`.
+  - You may need to run more than one command. For example, to edit the global
+  integration credentials both in and out of the `govuk::apps` namespace, you
+  need run both `$ bundle exec rake 'eyaml:edit[integration]'` and
+  `$ bundle exec rake 'eyaml:edit[integration,apps]'`.
+- Search for the `Access Key ID` in the result.
+
+Repeat the previous steps for [govuk-secrets/puppet_aws]:
+
+- Enter [govuk-secrets/puppet_aws].
+- Run the same command(s) again.
+- Search for the `Access Key ID` in the result.
+
+Keys may also be defined in more then one place. For example, the hieradata
+values `govuk::apps::support::aws_access_key_id` and
 `govuk::apps::support_api::aws_access_key_id` contain the same key.
+
 Keys may also be shared across AWS and Carrenza environments.
 
-If the key is not present in the hieradata, it might be present in the control panel of one of the 3rd party services, such as Fastly or Logit.
+If the key is not present in the hieradata, it might be present in the control
+panel of one of the 3rd party services, such as Fastly or Logit.
 
-### 3. Create and deploy new key
+> **Note**
+> If the key is not present in `govuk-secrets` nor in any other 3rd party
+> services, you can [remove the key].
```

Comment:
It might be good to make this a more general comment on what to do if you haven't managed to find the key. It could be that AWS says the key is in use, or that AWS says the key hasn't been used recently (or maybe the reader wouldn't get to this point if that's the case).

Reply:
It's possible to have a key in AWS that is not in use anywhere, even if the key has a `Last used` date. Someone might have removed because it was no longer needed and forgotten to remove it from AWS. If that's the case, the dev reading this guide would arrive at this point of the guide and not find the key anywhere. I wanted to make it clear that, in this case, it's safe to simply delete the key. I put it at the end of `### 3. Find where the key is used` because it's the very last possibility (we should first try and find the key) and I made it a note because so it stands out.

Reply:
I made this the last point of this step. No need to be a note.